src/serac/physics/solid_functional.hpp

```diff
@@ -130,28 +135,34 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
     state_.push_back(displacement_);
     state_.push_back(adjoint_displacement_);
 
+    if (calc_shape_ == ShapeDisplacement::On) {
+      state_.push_back(shape_displacement_);
+    }
+
     // Create a pack of the primal field and parameter finite element spaces
-    mfem::ParFiniteElementSpace*                                             test_space = &displacement_.space();
-    std::array<mfem::ParFiniteElementSpace*, sizeof...(parameter_space) + 2> trial_spaces;
+    mfem::ParFiniteElementSpace* test_space = &displacement_.space();
+
```

Comment:
The "3" magic number appears here, too. Maybe it's worth pulling this into a named constant?